ui/src/components/role/AddRole.js

```diff
@@ -152,10 +154,14 @@ class AddRole extends React.Component {
         this.delegateChanged = this.delegateChanged.bind(this);
         this.onSubmit = this.onSubmit.bind(this);
         this.expandSettings = this.expandSettings.bind(this);
+        this.toggleAuditEnabled = this.toggleAuditEnabled.bind(this);
         this.dateUtils = new DateUtils();
 
-        let role = {};
-        role['selfServe'] = false;
+        let role = {
```

Comment:
self serve is not used delete it and add audit enable directly to state it is best practice to be consistent in all components, try to do it like you did in addGroup

Reply:
i cleaned it